packages/@aws-cdk/aws-redshift-alpha/lib/private/database-query-provider/table.ts

```diff
@@ -1,34 +1,40 @@
 /* eslint-disable-next-line import/no-unresolved */
 import * as AWSLambda from 'aws-lambda';
+import { Column } from '../../table';
 import { executeStatement } from './redshift-data';
 import { ClusterProps, TableAndClusterProps, TableSortStyle } from './types';
-import { areColumnsEqual, getDistKeyColumn, getSortKeyColumns } from './util';
-import { Column } from '../../table';
+import { areColumnsEqual, getDistKeyColumn, getSortKeyColumns, makePhysicalId } from './util';
 
 export async function handler(props: TableAndClusterProps, event: AWSLambda.CloudFormationCustomResourceEvent) {
   const tableNamePrefix = props.tableName.prefix;
-  const tableNameSuffix = props.tableName.generateSuffix === 'true' ? `${event.RequestId.substring(0, 8)}` : '';
+  const tableNameSuffix = props.tableName.generateSuffix === 'true' ? `${event.StackId.substring(event.StackId.length - 12)}` : '';
   const tableColumns = props.tableColumns;
   const tableAndClusterProps = props;
   const useColumnIds = props.useColumnIds;
 
   if (event.RequestType === 'Create') {
     const tableName = await createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
-    return { PhysicalResourceId: tableName };
+    return {
+      PhysicalResourceId: makePhysicalId(tableName, tableAndClusterProps, event.RequestId),
+      Data: { TableName: tableName },
+    };
```

Comment:
Can you explain this change? Am I correct that it distinguishes the PhysicalResourceId from the table name so that when the table name is updated, we can keep the id intact? The Data section then gives us the ability to access the table name as we used to be able to?

Reply:
Yes, as this was the root cause of the issue. The `PhysicalResourceId` was the table name. So instead of updating the `PhyscialResourceId` (which would cause a new creation of the resource), we separate the 2 attributes, to persist the table data, whilst allowing the changing of the table name